pkg/reporter/reporter_mock.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package reporter
+
+import (
+	"context"
+	"sync"
+)
+
+var _ Reporter = (*MockReporter)(nil)
+
+type Request struct {
+	Name   string
+	Params []any
+}
+
+// MockReporter implements the reporter interface for mocking in unit tests.
+type MockReporter struct {
+	reqMu sync.Mutex
+	Reqs  []*Request
+
+	CreateStatusErr error
+	UpdateStatusErr error
+}
+
+// CreateStatus implements the CreateStatus function.
+func (m *MockReporter) CreateStatus(ctx context.Context, p *Params) error {
+	m.reqMu.Lock()
+	defer m.reqMu.Unlock()
+	m.Reqs = append(m.Reqs, &Request{
+		Name:   "CreateStatus",
+		Params: []any{*p},
```

Comment:
Careful, this is a shallow copy of p, so if p is modified outside of the call site, it'll show here.

Reply:
fixed